src/Quantum/Quantum.jl

```diff
@@ -84,17 +96,14 @@ Base.hcat(tns::TensorNetwork...) = reduce(hcat, tns)
 function Base.adjoint(tn::TensorNetwork{A}) where {A<:Quantum}
     tn = deepcopy(tn)
 
-    # TODO refactor internals
-    for i in siteinds(tn)
-        plug = i.meta[:plug]
-        i.meta[:plug] = if plug == :input
-            :output
-        elseif plug == :output
-            :input
-        else
-            # TODO throw error?
-        end
-    end
+    tmp = Dict((site, if dir === :in
+        :out
+    elseif dir === :out
+        :in
+    else
+        dir
+    end) => index for ((site, dir), index) in tn[:plug])
+    merge!(tn[:plug], tmp)
```

Comment:
```suggestion
    for (key, value) in tn[:plug]
        tn[:plug][key] = if value === :in
            :out
        elseif value === :out
            :in
        else
            value
        end
    end
```
In place replacement of `:in` to `:out` and vice versa